linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analysis.scala

```diff
@@ -312,8 +323,17 @@ object Analysis {
                 log(level, s"$verb from ${methodInfo.fullDisplayName}")
                 if (onlyOnce(level, methodInfo)) {
                   involvedClasses ++= methodInfo.instantiatedSubclasses
-                  loopTrace(methodInfo.calledFrom.lastOption)
+                  methodInfo.calledFrom.lastOption match {
+                    case Some(fromDispatch: FromDispatch) if sameMethod(methodInfo, fromDispatch) =>
+                      // avoid logging "dispatch from C.m" just after "called from C.m"
```

Comment:
I've been thinking about this quite a bit, and I think this is indeed the best approach to avoid excessive logs but still provide info when relevant.